packages/sdk/src/server/__tests__/server.test.ts

```diff
@@ -0,0 +1,646 @@
+import {
+  ServerClient, createClient,
+} from "../index";
+import fetchMock from "jest-fetch-mock";
+import { ClientCredentials } from "simple-oauth2";
+
+describe("ServerClient", () => {
+  let client: ServerClient;
+
+  beforeEach(() => {
+    fetchMock.resetMocks();
+  });
+
+  describe("createClient", () => {
+    it("should mock the createClient method and return a ServerClient instance", () => {
+      const params = {
+        publicKey: "test-public-key",
+        secretKey: "test-secret-key",
+      };
+
+      const client = createClient(params);
+      expect(client).toBeInstanceOf(ServerClient);
+    });
+  });
+
+  describe("makeRequest", () => {
+    it("should make a GET request successfully", async () => {
+      fetchMock.mockResponseOnce(
+        JSON.stringify({
+          data: "test-response",
+        }),
+        {
+          headers: {
+            "Content-Type": "application/json",
+          },
+        },
+      );
+
+      client = new ServerClient({
+        publicKey: "test-public-key",
+        secretKey: "test-secret-key",
+      });
+
+      const result = await client["makeRequest"]("/test-path", {
```

Comment:
The `makeRequest` method does not exist, right? There's the `_makeRequest` method, which is private so I'd suggest not testing it directly and just focus on the public interface of `ServerClient`.

Also, why do we have to do `client["makeRequest"]` instead of `client.makeRequest`? Is it because of mocking? Or because the method is private?

Reply:
I'd rather focus the tests on the public interface instead.